### PR TITLE
[WIP] Move cherry-picker to scm versioning via Git tags

### PR DESCRIPTION
**Pull request by webknjaz**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ before_install:
     TARGET_PKG: cherry_picker
   install:
   - &cd-to-project pushd "$TARGET_PKG"
-  - flit install
+  - &flit-install flit install --extras dev,test
```
Comment (Mariatta) [inline .travis.yml]: The command is 
`flit install --deps develop`

It will install dev and tests requirements.

Comment (webknjaz) [reply .travis.yml]: I tried this: see ~10 last commits :(

Comment (webknjaz) [reply .travis.yml]: I tried this: see ~10 last commits :(

**Comment by webknjaz** [comment]
@Mariatta for some reason it says "production" in log all the time: https://travis-ci.com/webknjaz/core-workflow/jobs/142033230#L471
No matter what command runs.

**Comment by webknjaz** [comment]
Okay, so here's what's happening: https://github.com/takluyver/flit/blob/fcdb3ee5009624863011570f3a9dd82cb86cacd7/flit/install.py#L263

It does reqs probe by pre-importing the module and then runs subinstaller to install "production" deps

**Comment by webknjaz** [comment]
I've failed to make it work because of the way flit extracts values from evaluated module :(
So it uses `module.__dict__.get('__version__', None)`, which works with already evaluated data, not triggering `__getattr__` stuff.

**Comment by Mariatta** [comment]
Thanks. Sorry it's not working out with flit.
Looking at how setuptools-scm works now, I think I prefer having explicit version info `__version__ = X.Y.Z`, as part of codebase, instead of deriving it from git tag.
Let me rethink of this workflow a bit more.
Thanks.

**Comment by webknjaz** [comment]
Well, setuptools-scm does work well on its own. The problem is with flit not installing dependencies on time (before trying to mess with imports).
Just to be clear: I've had it actually working great with `flit` when `setuptools-scm` was pre-installed before running `flit install`.

**Comment by hugovk** [comment]
Cherry Picker now lives in its own repo at https://github.com/python/cherry-picker

Please re-open there is still relevant.

**Pull request closed without merging**